python3Packages.click: fix SystemExit(1) when using click.File's stdin support

dotlambda opened the pull request:
This fixes upstream issue https://github.com/pallets/click/issues/2939 which e.g. breaks swh.
See also https://github.com/NixOS/nixpkgs/pull/454434#issuecomment-3485685749.
Should I target staging-next?

ZHF: https://github.com/NixOS/nixpkgs/issues/457852

closes https://github.com/NixOS/nixpkgs/pull/458890
cc @drupol who I can't request a review from (maybe they blocked me?)
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

drupol committed:
python3Packages.swh-scanner: disable tests failing on click >= 8.2.0

The patch is no longer needed because the respective test is now
disabled.

dotlambda committed:
python3Packages.click: fix SystemExit(1) when using click.File's stdin support

This fixes upstream issue https://github.com/pallets/click/issues/2939 which e.g. breaks swh.

dotlambda committed:
python3Packages.swh-core: 4.5.2 -> 4.6.0

Diff: https://gitlab.softwareheritage.org/swh/devel/swh-core/-/compare/v4.5.2...v4.6.0

Changelog: https://gitlab.softwareheritage.org/swh/devel/swh-core/-/tags/v4.6.0

dotlambda committed:
python3Packages.swh-web-client: unpin click

wolfgangwalther commented:
> who I can't request a review from (maybe they blocked me?)

took a long time to load, but ultimately worked.

dotlambda committed:
python3Packages.swh-scanner: fix meta

dotlambda committed:
python3Packages.swh-storage: 3.1.0 -> 4.1.0

Diff: https://gitlab.softwareheritage.org/swh/devel/swh-storage/-/compare/v3.1.0...v4.1.0

Changelog: https://gitlab.softwareheritage.org/swh/devel/swh-storage/-/tags/v3.1.1
           https://gitlab.softwareheritage.org/swh/devel/swh-storage/-/tags/v3.1.2
           https://gitlab.softwareheritage.org/swh/devel/swh-storage/-/tags/v3.2.0
           https://gitlab.softwareheritage.org/swh/devel/swh-storage/-/tags/v3.3.0
           https://gitlab.softwareheritage.org/swh/devel/swh-storage/-/tags/v4.0.0
           https://gitlab.softwareheritage.org/swh/devel/swh-storage/-/tags/v4.1.0

dotlambda committed:
python3Packages.swh-objstorage: 4.0.0 -> 5.1.0

Diff: https://gitlab.softwareheritage.org/swh/devel/swh-objstorage/-/compare/v4.0.0...v5.1.0

Changelog: https://gitlab.softwareheritage.org/swh/devel/swh-objstorage/-/tags/v4.1.0
           https://gitlab.softwareheritage.org/swh/devel/swh-objstorage/-/tags/v4.2.0
           https://gitlab.softwareheritage.org/swh/devel/swh-objstorage/-/tags/v5.0.0
           https://gitlab.softwareheritage.org/swh/devel/swh-objstorage/-/tags/v5.0.1
           https://gitlab.softwareheritage.org/swh/devel/swh-objstorage/-/tags/v5.1.0

dotlambda committed:
python3Packages.swh-shard: init at 2.2.0

dotlambda committed:
python3Packages.swh-scheduler: 3.1.0 -> 3.3.0

Diff: https://gitlab.softwareheritage.org/swh/devel/swh-scheduler/-/compare/v3.1.0...v3.3.0

Changelog: https://gitlab.softwareheritage.org/swh/devel/swh-scheduler/-/tags/v3.2.0
           https://gitlab.softwareheritage.org/swh/devel/swh-scheduler/-/tags/v3.3.0

drupol commented:
Thanks for making the patch ! Going to review this tomorrow.

I actually started to work on this too @ https://github.com/NixOS/nixpkgs/pull/458890 but your approach is better.

iedame commented:
Thank you for the patch! I will do some testing but it looks great.

drupol reviewed (approved):
Quickly reviewed from the phone, LGTM !

dotlambda commented:
> Should I target staging-next?

This question still hasn't been answered.

drupol commented:
Given the amount of rebuilds, I would say yes.
